src/radical/pilot/umgr/staging_input/default.py

```diff
@@ -338,12 +340,23 @@ def _handle_unit(self, unit, actionables):
             src = complete_url(src, src_context, self._log)
             tgt = complete_url(tgt, tgt_context, self._log)
 
-            if rpc.CREATE_PARENTS in flags:
-                copy_flags = rs.filesystem.CREATE_PARENTS
-            else:
-                copy_flags = 0
+            # Check if the src is a folder, if true
+            # add recursive flag if not already specified
+            if os.path.isdir(src.path):
+                if rs.filesystem.RECURSIVE not in flags:
+                    flags += [rs.filesystem.RECURSIVE]
 
-            saga_dir.copy(src, tgt, flags=copy_flags)
+            # Check if create parents flag is set,
+            # if not set it.
+            if rs.filesystem.CREATE_PARENTS not in flags:
+                flags += [rs.filesystem.CREATE_PARENTS]
```

Comment:
Why do we need the `CREATE_PARENTS` flag in RP if its always added anyway?

Reply:
If you are commenting that we don't need the recursive flag at the user API, I agree. This can be made the default.

Reply:
Well, I am not sure.  You changed semantics here I believe, on purpose, so its on you to argue for the API change :)

Reply:
CREATE_PARENTS sets up all the parent folders of an file/folder, IIUC. Shouldn't that be the default always? The above was a way of implementing it and making sure the user does not overwrite it. There might be other way of enforcing it though.